service-activator/service-activator.bal

```diff
@@ -0,0 +1,35 @@
+import ballerina/http;
+
+type ClaimRequest record {|
+    string customerId;
+    string claimantName;
+    string policyNumber;
+    decimal claimAmount;
+    string accidentType;
+    string claimLocation;
+    string claimDate;
+|};
+
+type Claim record {|
+    *ClaimRequest;
+    string[] remarks;
+|};
+
+type ClaimHistory record {|
+    string customerId;
+    Claim[] claims;
+|};
```

Comment:
Isn't the field `customerId` found too many times inside `ClaimHistory`? (First directly in `ClaimHistory` and then again inside every `Claim`)

Reply:
Yes, there is duplication. But I couldn't able to think of a simple way to extract Claim details from firebase.

Reply:
I had to sacrifice the duplication of customerId to reduce the code complexity.